doc/design/008-state-proof-pluggable-parsing/README.md

```diff
@@ -0,0 +1,127 @@
+## Legend
+There are some types of requests to Nodes in the Pool allowing to use StateProof optimization in Client-Node communication.
+Instead of sending requests to all nodes in the Pool, a client can send a request to a single Node and expect StateProof signed by BLS multi-signature.
+
+BLS multi-signature (BLS MS) guaranties that there was a consensus of Nodes which signed some State identified by State RootHash.
+StateProof (SP) is small amount of data which allows to verify particular values against RootHash.
+Combination of BLS MS and SP allows clients to be sure about response of single node is a part of State signed by enough Nodes. 
+
+## Goals
+Libindy allows to create and send custom transaction via plugged interface.
```

Comment:
> Libindy allows to create and send custom transaction via plugged interface.

Not sure that it is 100% correct. More correct version:
1. Indy Node allows to extend supported requests with pluggins
2. libindy allows to send any request to node (we don't need plugins here), but some requests can be significantly optimized with signed state proofs.